repair/row_level.cc

```diff
@@ -3229,11 +3231,16 @@ future<> repair_service::cleanup_history(tasks::task_id repair_id) {
 }
 
 future<> repair_service::load_history() {
-    co_await get_db().local().get_tables_metadata().for_each_table_gently(coroutine::lambda([&] (table_id table_uuid, lw_shared_ptr<replica::table> table) -> future<> {
-        auto shard = unsigned(table_uuid.uuid().get_most_significant_bits()) % smp::count;
+    co_await get_db().local().get_tables_metadata().parallel_for_each_table(coroutine::lambda([&] (table_id table_uuid, lw_shared_ptr<replica::table> table) -> future<> {
+        auto str_uuid = table_uuid.to_sstring();
+        utils::crc32 crc;
+        crc.process(reinterpret_cast<const uint8_t*>(str_uuid.data()), str_uuid.size());
+        auto shard = crc.get() % smp::count;
```

Comment:
I do not like the extra crc checksum for choosing the shard. 

Does (get_most_significant_bits() ^ get_least_significant_bits()) % smp::count help your case?

Reply:
The 128 bits uuid has enough entropy . We can just fold them to a shard id which is uint32_t. 

uint32_t uuid_to_shard(uint64_t x, uint64_t y) {
    uint32_t sum;
    do {
        sum ^= uint32_t(x);
        x = x >> 16;
    } while (x);
    do {
        sum ^= uint32_t(y);
        y = y >> 16;
    } while (y);
    return sum;
}

Reply:
See here: https://github.com/scylladb/scylladb/pull/17049